Quadriga/src/main/java/edu/asu/spring/quadriga/web/settings/WebsiteAboutEditController.java

```diff
@@ -0,0 +1,75 @@
+package edu.asu.spring.quadriga.web.settings;
+
+import java.security.Principal;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import edu.asu.spring.quadriga.aspects.annotations.AccessPolicies;
+import edu.asu.spring.quadriga.aspects.annotations.CheckedElementType;
+import edu.asu.spring.quadriga.aspects.annotations.ElementAccessPolicy;
+import edu.asu.spring.quadriga.aspects.annotations.InjectProject;
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.publicwebsite.IAboutTextManager;
+import edu.asu.spring.quadriga.service.workbench.IRetrieveProjectManager;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+/**
+ * This controller manages public website's about page of a project. Information
+ * of title and description is editable.
+ *
+ * @author Rajat Aggarwal
+ *
+ */
+
+@Controller
+public class WebsiteAboutEditController {
+
+	@Autowired
+	private IRetrieveProjectManager projectManager;
+
+	@Autowired
+	private IAboutTextManager aboutTextManager;
+
+	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 0, userRole = {
+			RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+		@RequestMapping(value = "auth/workbench/projects/{ProjectUnixName}/settings/editabout", method = RequestMethod.GET)
```

Comment:
Since this is part of the authenticated part of Quadriga, you should have the project id in the path not the unix name. So you have to retrieve the project by its id yourself and can't use the InjectProject annotation.